### PR TITLE
feat(icon-pins): #MED-297 add pin icon and pin right

### PR DESCRIPTION
**Pull request by niko-amosse**
## Describe your changes

We create a pin provider in order to fetch pins evry where. We create pin rigth (to manage) and display pin icon.

Display with pin rigth : 
image

Display without : 
image

## Checklist tests
* check that requests to manage are blocked (back) when user doesnt have right
* check that pin icon is only displayed when user have right
* check all actions around pins : create, set, delete (on differente pages)

## Issue ticket number and link

[MED-297](https://jira.support-ent.fr/browse/MED-297)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [X] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: frontend/src/hooks/useExternalResource.ts [hunk 1]
```diff
@@ -33,12 +36,24 @@ export const useExternalResource = () => {
...
+          (externalResource: ExternalResource) => ({
+            ...externalResource,
+            is_pinned: pins.some(
+              (pin: Pin) =>
+                pin?.id == externalResource?.id &&
+                pin.source === "fr.openent.mediacentre.source.GAR",
+            ),
+          }),
+        );
+      }
```
Comment (niko-amosse) [inline frontend/src/hooks/useExternalResource.ts:53]: when pins are modified we refetch all data to check pin state

File: frontend/src/hooks/useFavorite.ts [hunk 2]
```diff
@@ -20,9 +23,18 @@ export const useFavorite = () => {
...
       }));
+      if (pins) {
+        favoriteData = favoriteData.map((favorite: Favorite) => ({
+          ...favorite,
+          is_pinned: pins.some(
+            (pin: Pin) =>
+              pin?.id == favorite?.id && pin.source === favorite?.source,
+          ),
+        }));
+      }
```
Comment (niko-amosse) [inline frontend/src/hooks/useFavorite.ts:34]: idem on remet à jour le champ is_pinned à chaque update de pins

File: frontend/src/hooks/useGlobal.ts [hunk 3]
```diff
@@ -17,9 +20,17 @@ export const useGlobal = () => {
...
         favorite: favorites.some((fav: Favorite) => fav._id === global._id),
       }));
+      globalData = globalData.map((global: GlobalResource) => ({
+        ...global,
+        is_pinned: pins.some(
+          (pin: Pin) =>
+            pin?.id === global?._id &&
+            pin.source === "fr.openent.mediacentre.source.GAR",
+        ),
+      }));
```
Comment (niko-amosse) [inline frontend/src/hooks/useGlobal.ts:30]: idem

File: frontend/src/hooks/useSearch.ts [hunk 4]
```diff
@@ -30,17 +33,45 @@ export const useSearch = (query: any) => {
...
+        }));
+        garResources = garResources.map((garResource: Resource) => ({
+          ...garResource,
+          is_pinned: pins.some(
+            (pin: Pin) =>
+              pin?.id == garResource?.id &&
+              pin.source === "fr.openent.mediacentre.source.GAR",
+          ),
+        }));
+      }
```
Comment (niko-amosse) [inline frontend/src/hooks/useSearch.ts:66]: idem

File: frontend/src/hooks/useSignet.ts [hunk 5]
```diff
@@ -63,11 +66,21 @@ export const useSignet = () => {
...
+    if (pins) {
+      signetsData = signetsData.map((signet: Signet) => ({
+        ...signet,
+        is_pinned: pins.some(
+          (pin: Pin) =>
+            pin?.id == signet?.id &&
+            pin.source === "fr.openent.mediacentre.source.Signet",
+        ),
+      }));
+    }
```
Comment (niko-amosse) [inline frontend/src/hooks/useSignet.ts:78]: idem

File: frontend/src/components/pins-carousel/PinsCarousel.tsx [hunk 6]
```diff
@@ -10,17 +10,18 @@ import { useTranslation } from "react-i18next";
 import { PinsCarouselCard } from "./pins-carousel-card/PinsCarouselCard";
 import "@splidejs/react-splide/css";
 import "./PinsCarousel.scss";
-import { Pin } from "~/model/Pin.model";
+import { usePinProvider } from "~/providers/PinProvider";
```
Comment (ThomasSevagen) [inline frontend/src/components/pins-carousel/PinsCarousel.tsx:13]: ça fait plaisir :)

File: frontend/src/providers/PinProvider.tsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
+import React, {
```
Comment (ThomasSevagen) [inline frontend/src/providers/PinProvider.tsx:1]: Excellente définition de provider, à voir si dans la dépendance de ton useMemo, tu as vraiment besoin de mettre refetchPins. Si ça te crée pas de render non voulu tu peux laisser

**Review by ThomasSevagen — APPROVED** [review]

File: frontend/src/components/modals/create-pins/CreatePins.tsx [hunk 8]
```diff
@@ -102,7 +102,7 @@ export const CreatePins: React.FC<CreatePinsProps> = ({ refetch }) => {
               onError={({ currentTarget }) => {
                 currentTarget.onerror = null;
                 currentTarget.src =
-                  "/mediacentre/public/img/no-image-resource.svg";
+                  "/mediacentre/public/img/no-image-resource.png";
```
Comment (Rseuret) [inline frontend/src/components/modals/create-pins/CreatePins.tsx:105]: constant

File: frontend/src/components/modals/edit-pins/EditPins.tsx [hunk 9]
```diff
@@ -106,7 +106,7 @@ export const EditPins: React.FC<EditPinsProps> = ({ refetch }) => {
               onError={({ currentTarget }) => {
                 currentTarget.onerror = null;
                 currentTarget.src =
-                  "/mediacentre/public/img/no-image-resource.svg";
+                  "/mediacentre/public/img/no-image-resource.png";
```
Comment (Rseuret) [inline frontend/src/components/modals/edit-pins/EditPins.tsx:109]: constant

**Review by Rseuret — APPROVED** [review]

**Pull request merged**